restart_process/README.md

```diff
@@ -1,10 +1,123 @@
+The call above looks just like the initial `docker_build` call except for one added parameter, `entrypoint` (in this example, `/go/bin/foo`). This is the command that you want to run on container start and _re-run_ on Live Update.
+
+### Function Signature
+```python
+def docker_build_with_restart(ref: str, context: str,
+    entrypoint: Union[str, List[str]],
+    live_update: List[LiveUpdateStep],
+    base_suffix: str = '-base',
+    restart_file: str = '/.restart-proc',
+    **kwargs
+):
+    """Args:
+      ref: name for this image (e.g. 'myproj/backend' or 'myregistry/myproj/backend'); as the parameter of the same name in docker_build
+      context: path to use as the Docker build context; as the parameter of the same name in docker_build
+      entrypoint: the command to be (re-)executed when the container starts or when a live_update is run
+      live_update: set of steps for updating a running container; as the parameter of the same name in docker_build
+      base_suffix: suffix for naming the base image, applied as {ref}{base_suffix}
+      restart_file: file that Tilt will update during a live_update to signal the entrypoint to rerun
+      **kwargs: will be passed to the underlying `docker_build` call
+    """
+```
+## Unsupported Cases
+This extension does NOT support process restarts for:
+- Images run in Docker Compose resources
+- `custom_build`
+
+Run into a bug? Need a use case that we don't yet support? Let us know--[open an issue](https://github.com/windmilleng/tilt-extensions/issues) or [contact us](https://tilt.dev/contact).
+
+## What's Happening Under the Hood
+*If you're a casual user/just want to get your app running, you can stop reading now. However, if you want to dig deep and know exactly what's going on, or are trying to debug weird behavior, read on.*
+
+This extension wraps commands in `tilt-restart-wrapper`, which makes use of [`entr`](https://github.com/eradman/entr/)
+to run arbitrary commands whenever a specified file changes. Specifically, we override the container's entrypoint with the following:
+
+```
+/tilt-restart-wrapper --watchfile='/.restart-proc' <entrypoint>
+```
+
+This invocation says:
+- when the container starts, run <entrypoint>
+- whenever the `/.restart-proc` file changes, re-execute <entrypoint>
+
+We also set the following as the last `live_update` step:
+```python
+run('date > /.restart-proc')
+```
+
+Because `tilt-restart-wrapper` will re-execute the entrypoint whenever `/.restart-proc'` changes, the above `run` step will cause the entrypoint to re-run.
+
+#### Provide `tilt-restart-wrapper`
+For this all to work, the `entr` binary must be available on the Docker image. The easiest solution would be to call e.g. `apt-get install entr` in the Dockerfile, but different base images will have different package mangers; rather than grapple with that, we've made a statically linked binary available on Docker image: [`tiltdev/entr`](https://hub.docker.com/repository/docker/tiltdev/entr).
```

Comment:
```suggestion
For this all to work, the `entr` binary must be available on the Docker image. The easiest solution would be to call e.g. `apt-get install entr` in the Dockerfile, but different base images will have different package managers; rather than grapple with that, we've made a statically linked binary available on Docker image: [`tiltdev/entr`](https://hub.docker.com/repository/docker/tiltdev/entr).
```